lua/luasnip/nodes/snippet.lua

```diff
@@ -228,9 +245,10 @@ local function init_snippet_context(context, opts)
 	end
 	effective_context.trig_matcher = engine(effective_context.trigger)
 
-	effective_context.condition = context.condition
-		or opts.condition
-		or true_func
+	effective_context.context_matcher = context.context_matcher
+		or opts.context_matcher
```

Comment:
Ah, the `or opts.condition` above was for backwards compatibility, `context_matcher` should only be set from `context`.
(options from context relate to stuff happening before the expansion of the snippet, where as fields from opts influence behaviour after expansion)

Reply:
fix it.